daft/dataframe/chroma_data_sink.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+from daft import DataType
+from daft.io import DataSink
+from daft.io.sink import WriteResult
+from daft.recordbatch import MicroPartition
+from daft.schema import Schema
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+    from types import ModuleType
+
+
+class ChromaDataSink(DataSink[None]):
+    """WriteSink for writing data to a Chroma collection."""
+
+    def _import_chromadb(self) -> ModuleType:
+        try:
+            import chromadb
+
+            return chromadb
+        except ImportError:
+            raise ImportError("chromadb is not installed. Please install chromadb using `pip install chromadb`")
+
+    def __init__(
+        self,
+        persist_directory: str,
+        collection_name: str,
+        embedding_column: str | None = None,
+        document_column: str = "document",
+        id_column: str = "id",
+        metadata_column: str | None = None,
+        mode: Literal["create", "append"] = "append",
+    ):
+        chromadb = self._import_chromadb()
+        client = chromadb.PersistentClient(path=persist_directory)
+        self._persist_directory = persist_directory
+        self._collection_name = collection_name
+        self._embedding_column = embedding_column
+        self._document_column = document_column
+        self._id_column = id_column
+        self._metadata_column = metadata_column
+        self._mode = mode
+
+        from chromadb.utils import embedding_functions
+
+        sentence_transformer_ef = embedding_functions.SentenceTransformerEmbeddingFunction(
+            model_name="all-MiniLM-L6-v2"
+        )
+
+        if mode == "create":
+            try:
+                client.delete_collection(name=collection_name)
+            except Exception:
+                pass
+            client.create_collection(
+                name=collection_name, configuration={"embedding_function": sentence_transformer_ef}
+            )
+        else:
+            client.get_or_create_collection(name=collection_name)
```

Comment:
This is super cool. Why not pass either the configuration or embedding_function as a parameter?

Reply:
Makes sense!

For full disclosure I hacked this up after reading Chroma documentation for a few minutes to see how easy it would be to write a data sink for it. I realize now that the PR needs a bit more love. In particular, I just learned that Chroma `PersistentClient`s are intended for local storage only and for distributed workloads we'd use a Chroma `HTTPClient` 😅 

I'll add the ability for users to select either depending on their use case.